### ISSUE TITLE
[Feature] Don't clear the console

### ISSUE THREAD

**Issue by karlwilbur** [sysgears/mochapack#26]
- [x ] I'd be willing to implement this feature
- [ ] This feature can already be implemented through a plugin

**Describe the user story**

I am using my console. It has valuable historical information. Deleting my valuable information is not wanted. Don't clear the console.

**Describe the solution you'd like**

When I run `mochapack`, it doesn't clear the console. Rather it just runs in the existing screen as is expected in a Unix-like environment.

**Describe the drawbacks of your solution**
Some people might like to throw away their console history. While I can certainly see the value in that, most terminals have an option to do that at the user's discretion. 

**Describe alternatives you've considered**
Exploring another package that doesn't clear the console. Forking this repo and removing the `clearConsole` method form `src/runner/testRunnerReporter.js:58` along with all calls to it.

**Additional context**
It would be reasonable to create this as a option.

**Comment by karlwilbur** [comment]
After reading through more of the code, it seems that this can me accomplished by setting `interactive` to `false`, that is:
```
  "scripts": {
    ..
    "test-js": "mochapack --interactive=false  ..."
  }
```
This request can be closed. I am not closing it myself as there might be a better, more intuitive way to make this happen; perhaps, minimally, a blurb in the docs somewhere. I'll leave that up to the maintainers to determine. If you'd like my help with this, just say the word; glad to contribute.

**Comment by vlasenko** [comment]
@karlwilbur I agree with you. I'm not sure whether adding this into the docs, would be of much help, but it certainly doesn't hurt. Perhaps we should turn on interactive mode only during watch and during normal test run it should be off by default. What do you think?

**Comment by karlwilbur** [comment]
> Perhaps we should turn on interactive mode only during watch and during normal test run it should be off by default.

Yes, I would like that. However, making a change to defaults might be jarring to current users. Such changes should be made with caution.

As for "interactive" mode as a thing, I'm not sure what that even means for this tool. I'm still new to `mochapack` and not sure about the feature set yet. How would one "interact" in interactive mode? Why does that necessitate clearing the console history?

There are several other existing tools that watch for file changes and act on them (`guard` in Ruby for running tests and `livereload` integration, etc.; `webpack-dev-server` for compiling and HMR) which do not wipe the terminal.  

I think that there is just a philosophical difference here. To me, clearing the console is a user choice which should not be made the the tools that the user is running but rather them user themselves when they determine it is appropriate. The only other tool that I have seen do this is `react-scripts start` which I quickly `eject`ed and overhauled because I considered it extremely rude of them to just delete my terminal history without asking. 

Possibly just a config option to enable/disable clearing the console. I realize that some devs simply just don't care about their terminal history and want to make the "noise" of the console to just go away (I've worked along side such developers so that viewpoint is very much understood by me). For those devs, I wouldn't want to take away that convenience.

**Comment by vlasenko** [comment]
@karlwilbur My personal preference is to not clear the console by default and have a cli option to clear the console, instead.

**Comment by karlwilbur** [comment]
@vlasenko Mine too. Should I re-create a fork and take a stab at an implementation? I'll be unable to do anything for a few days but maybe early next week?

Would `--clear-console=true` be reasonable? It can default to `false`.

**Comment by vlasenko** [comment]
@karlwilbur Sounds reasonable to me, no rush, take your time

**Comment by andykais** [comment]
I would still argue that the console should clear by default on watch. This is how the typescript compiler behaves for example. (see the [--preserveWatchOutput](https://www.typescriptlang.org/docs/handbook/compiler-options.html) flag in the typescript handbook). I would suggest simply setting a default `--interactive=false` unless mochapack is invoked with `--watch`.

Also just for those curious (@karlwilbur), I believe the webpack `--interactive` flag derivative of the debian [DEBIAN_FRONTEND](https://serverfault.com/questions/48724/100-non-interactive-debian-dist-upgrade) variable.

**Comment by karlwilbur** [comment]
@vlasenko Finally got caught up on things; taking a vacation sure does create a backlog.

I opted to go with `clear-terminal`, since "terminal" is more technically correct.

The tests pass on my PR and I have tested it in my project by running my project's tests.

Having my `package.json` like this:
```
  "scripts": {
    ...
    "test": "mochapack --webpack-config config/webpack/test.js --require spec/javascript/setup.js spec/javascript/*.spec.js"
  }
```
...works as I would like (not deleting my terminal history).

This way works as before (clearing and purging the current terminal):
```
  "scripts": {
    ...
    "test": "mochapack --clear-terminal --webpack-config config/webpack/test.js --require spec/javascript/setup.js spec/javascript/*.spec.js"
  }
```

I have confirmed both work as expected in my project. Let me know if there is anything else I can do.
